x/concentrated-liquidity/tick.go

```diff
@@ -191,6 +191,34 @@ func validateTickRangeIsValid(tickSpacing uint64, lowerTick int64, upperTick int
 	return nil
 }
 
+// roundTick takes a tick and determines if multiple ticks can represent the same price as the provided tick. If so, it
+// rounds that tick up to the largest tick that can represent the same price that the original tick corresponded to. If one of
+// the two ticks are modified, we re-validate the tick range to ensure that the tick range is still valid.
+//
+// i.e. the provided tick is -161795100. With our precision, this tick correlates to a sqrtPrice of 0.000000001414213563
+// the first tick (given our precision) that is able to represent this price is -161000000, so we use this tick instead.
+//
+// This really only applies to very small tick values, as the increment of a single tick continues to get smaller as the tick value gets smaller.
+func roundTick(lowerTick, upperTick int64, priceTickLower, priceTickUpper sdk.Dec, tickSpacing uint64) (int64, int64, error) {
```

Comment:
No, its a little confusing though. The "roundDown" in `PriceToTickRoundDown` is referring to tick spacing rounding. The `PriceToTickRoundDown` is rounding us "up" to the largest tick that represents this sqrtPrice, but rounds it down when taking into consideration tick spacing. So the word "rounding" and its direction depends on what lens you are viewing it from. From the scale of ticks, its rounding up since if we give it a smaller tick that has other ticks larger than it that represent the same sqrtPrice, it will return the large tick. But from the tick spacing perspective, it will round down to ensure that tick being returned is a multiple of the tick spacing of the pool. Please lmk if that makes sense 

CC @p0mvn

Reply:
This makes sense to me, thanks for the breakdown. Would be valuable to update docs with this context